### DISCUSSION TITLE
Make the integration work "out of the box"

### DISCUSSION THREAD

**Discussion by exodrifter** [exodrifter/unity-python#36] [Ideas]
Right now, the Python standard library is not included in the build. The reason for this is because the IronPython library only exposes a method for loading python libraries from the disk (as far as I know). We can accomplish this by putting the Python library in the `StreamingAssets` folder, but this is a security problem since the library can be trivially edited by any user to do whatever they might want. This is the main reason for why the included python library in unity-python doesn't work in builds right out of the box.

Several issues are affected by this problem:
* The additional steps are not very clear and make the library difficult to use since it doesn't work "out of the box" #10
* It's not clear how to add additional libraries to the project #4, #8
* Adding additional libraries might require the addition of more .NET DLLs to the project in order for the build to work #11

I did find an approach that looks appealing recently... [It might be possible to compile the Python libraries into DLLs](https://web.archive.org/web/20190915142110/http://community.sharpdevelop.net/blogs/mattward/archive/2010/03/16/CompilingPythonPackagesWithIronPython.aspx).

If this works the way I think it does, then I can make an interface for installing Python libraries which fetches the package from `pip`, compiles it into a DLL, and then adds it to your project. This would fix all of the aforementioned issues.

However, it would be best to keep in mind that Python 2 is being deprecated on January 1st, 2020 with `pip` to follow suit shortly after. I have never tested the library as it stands today with Python 3 (and I don't have any reason to believe it will work). So, perhaps this work will have to wait until the library is ported to work with Python 3.

**Comment by Lingfeng158** [comment]
Thank you very much! Your work is much needed! May I know the ETA of this integration update?

**Comment by exodrifter** [comment]
@Lingfeng158 I work on this in my free time so I'm unable to give an ETA. I've been thinking about this issue for quite a while, thank you for expressing interest. :heart:
